spec/views/events/index.html.erb_spec.rb

```diff
@@ -33,4 +33,25 @@
     expect(rendered).to have_link(href: edit_event_path(@event))
     expect(rendered).to have_link(href: event_path(@event), class: 'btn-danger')
   end
+
+  it "should not display apply button if application deadline is over" do
+    @event.application_deadline = Date.yesterday
+    assign(:events, [@event])
+    [:pupil, :coach, :organizer].each do |role|
```

Comment:
This test should ensure that the apply button is hidden if the deadline is over (independent of the role of the user). But you are right if there exists test that guarantee that coaches and organizers can not apply that is a little bit redundant.